mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -19,110 +19,446 @@
+        fi
+        if [ -n "${k_pending_merge_commit}" ] && [ "${k_new_pending_merge_commit}" != "${k_pending_merge_commit}" ]; then
+            # the new pending merge commit is different. Apply the old one.
+            echo "Cherry-picking k8s.io/kubernetes merge  ${k_pending_merge_commit}: $(git show --format="%s" -q ${k_pending_merge_commit})."
+            local date=$(git show --format="%aD" -q ${k_pending_merge_commit}) # author and committer date is equal for PR merges
+            git reset -q $(GIT_COMMITTER_DATE="${date}" GIT_AUTHOR_DATE="${date}" git-with-user commit-tree -p ${dst_merge_point_commit} -p HEAD -m "$(git show --format='%B' -q ${k_pending_merge_commit}; echo; echo "Kubernetes-commit: ${k_pending_merge_commit}")" "HEAD^{tree}")
+            fix-godeps "${deps}" "${is_library}" ${dst_needs_godeps_update}
+            dst_needs_godeps_update=false
+            dst_merge_point_commit=$(git rev-parse HEAD)
+        fi
+        k_pending_merge_commit="${k_new_pending_merge_commit}"
+
+        # is it a merge or a single commit on the mainline to apply?
+        if ! git show -q ${f_mainline_commit} | grep -q "^Merge: "; then
+            echo "Cherry-picking k8s.io/kubernetes commit ${k_mainline_commit}: $(git show --format="%s" -q ${f_mainline_commit})."
+
+            # reset Godeps.json?
+            if [ -n "$(godep-changes ${f_mainline_commit})" ]; then
+                reset-godeps ${f_mainline_commit}^
+                dst_needs_godeps_update=true
+            fi
+
+            # finally cherry-pick
+            if ! git-with-user cherry-pick --keep-redundant-commits ${f_mainline_commit} >/dev/null; then
+                echo
+                show-working-dir-status
+                return 1
+            fi
+
+            # if there is no pending merge commit, update Godeps.json because this could be a target of tag
+            if [ -z "${k_pending_merge_commit}" ]; then
+                fix-godeps "${deps}" "${is_library}" ${dst_needs_godeps_update}
+                dst_needs_godeps_update=false
+                dst_merge_point_commit=$(git rev-parse HEAD)
+            fi
+        else
+            local f_mainline_merge=${f_mainline_commit}
+
+            # find **latest** (in the sense of least distance to ${f_mainline_merge}) common ancestor of both parents of ${f_mainline_merge}.
+            # If the PR had no merge commits, this is the actual fork point. If there have been merges, we will get the
+            # latest of those merges. This will potentially drop commit history, but no actual changes.
+            local f_latest_branch_point_commit=$(git merge-base --octopus ${f_mainline_merge}^1 ${f_mainline_merge}^2)
+            if [ -z "${f_latest_branch_point_commit}" ]; then
+                echo "Not branch point found for PR merged through ${f_mainline_merge}. Odd."
+                return 1
+            fi
+
+            # start cherry-picking with latest merge, squash everything before
```

Comment:
Could you add a comment that this shouldn't happen anymore, because we ban merge commits in PRs in kube?

Reply:
done